pycket/pycket-lang/zoTransform.rkt

```diff
@@ -0,0 +1,581 @@
+                                           (if (not multiple-args?)
+                                               (map (lambda (x) (symbol->string (gensym))) (range num-args))
+                                               (list (symbol->string (gensym))))]
+                                          
+                                          [arg-mapping
+                                           (if (not multiple-args?)
+                                               (map (lambda (arg) (hash* 'lexical arg)) symbols-for-formals)
+                                               (hash* 'lexical (car symbols-for-formals)))] ;; this is ugly
+                                          [body (to-ast-single (lam-body clause)
+                                                               toplevels
+                                                               (append symbols-for-formals localref-stack))]
+                                          )
+                                     (hash* 'lambda arg-mapping
+                                            'body body))))
+                             clauses))))
+    
+
+(define (handle-install-value body-form toplevels localref-stack)
+  ;; Runs rhs to obtain count results, and installs them into existing
+  ;; slots on the stack in order, skipping the first pos stack positions.
+  (let* [(count (install-value-count body-form))
+         (pos (install-value-pos body-form))
+         (boxes? (install-value-boxes? body-form))
+         (rhs (install-value-rhs body-form))
+         (body (install-value-body body-form))
+         (count-lst (range count))
+
+         (binding-list (map (λ (c) (string-append "inst-val" (number->string c))) count-lst))
+         
+         (box-positions (map (λ (p) (+ p pos)) count-lst))
+         ]
+    (begin
+      ;; setting the boxes that let-void has put in the stack
+      (for ([i box-positions]) (set-box! (list-ref localref-stack i) (list-ref binding-list (- i pos))))
+      ;; producing json for pycket
+      (hash*
+       ;; let-bindings <- [count] {eval rhs}
+       'let-bindings (list (list binding-list
+                                 (to-ast-single rhs toplevels localref-stack)))
+       ;; let-body <- body
+       'let-body (to-ast-single body toplevels localref-stack)))))
+
+(define (handle-list list-form toplevels localref-stack)
+  ;; currently don't know how we can have a list as a body form,
+  ;; but experiments show that we could have some '()s as arguments to an application
+  ;; pycket seems to reflect this as source-name : "null"
+  (if (not (null? list-form))
+      (error 'handle-list "INVESTIGATE... we seem to have a (not null) list as a single body form in the byte-code")
+      (hash* 'source-name "null")))
+  ;; (map (lambda (form) (to-ast-single form toplevels localref-stack)) list-form))
+
+;; stack : (listof symbol?/prefix?/hash?)
+(define (to-ast-single body-form toplevels localref-stack)
+  (begin
+    (if DEBUG
```

Comment:
But I want it to produce an error to notify that there is an interesting case to investigate. Producing void with `when` would make it harder to detect.